archive/phar_without_stub.ksy

```diff
@@ -0,0 +1,217 @@
+    doc: The contents of each file in the archive (possibly compressed, as indicated by the file's flags in the manifest). The files are stored in the same order as they appear in the manifest.
+  - id: signature
+    type: signature
+    size-eos: true
+    if: manifest.flags.has_signature
+    doc: |
+      The archive's signature - a digest of all archive data before the signature itself.
+      
+      Note: Almost all of the available "signature" types are actually hashes, not signatures, and cannot be used to verify that the archive has not been tampered with. Only the OpenSSL signature type is a true cryptographic signature.
+enums:
+  signature_type:
+    0x1:
+      id: md5
+      -orig-id: PHAR_SIG_MD5
+      doc: Indicates an MD5 hash.
+    0x2:
+      id: sha1
+      -orig-id: PHAR_SIG_SHA1
+      doc: Indicates a SHA-1 hash.
+    0x4:
+      id: sha256
+      -orig-id: PHAR_SIG_SHA256
+      doc: Indicates a SHA-256 hash. Available since API version 1.1.0, PHP_Archive 0.12.0 and phar extension 1.1.0.
+    0x8:
+      id: sha512
+      -orig-id: PHAR_SIG_SHA512
+      doc: Indicates a SHA-512 hash. Available since API version 1.1.0, PHP_Archive 0.12.0 and phar extension 1.1.0.
+    0x10:
+      id: openssl
+      -orig-id: PHAR_SIG_OPENSSL
+      doc: |
+        Indicates an OpenSSL signature. Available since API version 1.1.1. This type is not documented in the phar extension's documentation of the phar format.
+        
+        Note: In older versions of the phar extension, this value was used for an undocumented and unimplemented "PGP" signature type (`PHAR_SIG_PGP`).
+types:
+  serialized_value:
+    seq:
+      - id: raw
+        size-eos: true
+        doc: The serialized value, as a raw byte array.
+    instances:
+      parsed:
+        pos: 0
+        type: php_serialized_value
+        doc: The serialized value, parsed as a structure.
+  file_flags:
+    seq:
+      - id: value
+        type: u4
+        doc: The unparsed flag bits.
+    instances:
+      permissions:
+        value: value & 0x1ff
+        -orig-id: PHAR_ENT_PERM_MASK
+        doc: The file's permission bits.
+      zlib_compressed:
+        value: (value & 0x1000) != 0
+        -orig-id: PHAR_ENT_COMPRESSED_GZ
+        doc: Whether this file's data is stored using zlib compression.
+      bzip2_compressed:
+        value: (value & 0x2000) != 0
+        -orig-id: PHAR_ENT_COMPRESSED_BZ2
+        doc: Whether this file's data is stored using bzip2 compression.
+  file_entry:
+    seq:
+      - id: filename_size
+        type: u4
+        doc: The size of the file name, in bytes.
+      - id: filename
+        size: filename_size
+        doc: The name of this file. If the name ends with a slash, this entry represents a directory, otherwise a regular file. Directory entries are supported since phar API version 1.1.1. (Explicit directory entries are only needed for empty directories. Non-empty directories are implied by the files located inside them.)
```

Comment:
Is it a `str`?

Reply:
Not really. The phar format doesn't specify an encoding for file names, and the PHP phar extension's API lets you use arbitrary byte sequences as file names (because PHP's strings are plain bytes with no specific encoding). In practice, almost all phar filenames are ASCII, but the spec shouldn't assume that.

Reply:
Thanks for the info. Could you add it into `doc`?